mcp/mastra/verticals/commute/tools.ts

```diff
@@ -0,0 +1,517 @@
+  inputSchema: z.object({
+    origin: z.string().describe('Starting point as address or coordinates'),
+    destination: z.string().describe('Destination as address or coordinates'),
+    searchQuery: z.string().describe('What to search for (e.g., "EV charging station", "coffee shop", "gas station")'),
+    maxResults: z.number().default(5).describe('Maximum number of results to return (1-20)'),
+  }),
+  outputSchema: z.object({
+    places: z.array(
+      z.object({
+        name: z.string(),
+        address: z.string(),
+        location: z.object({
+          lat: z.number(),
+          lng: z.number(),
+        }),
+        rating: z.number().optional(),
+        userRatingsTotal: z.number().optional(),
+        description: z.string().optional(),
+        types: z.array(z.string()).optional(),
+      }),
+    ),
+    routeInfo: z.object({
+      origin: z.string(),
+      destination: z.string(),
+    }),
+  }),
+  execute: async ({ origin, destination, searchQuery, maxResults }) => {
+    const apiKey = getApiKey();
+
+    const directionsParams = new URLSearchParams({
+      origin,
+      destination,
+      key: apiKey,
+    });
+
+    const directionsUrl = `${GOOGLE_MAPS_API_BASE}/directions/json?${directionsParams.toString()}`;
+    const directionsResponse = await fetch(directionsUrl);
+    const directionsData = await directionsResponse.json();
+
+    if (directionsData.status !== 'OK') {
+      throw new Error(
+        `Directions API error: ${directionsData.status} - ${directionsData.error_message || 'Unknown error'}`,
+      );
+    }
+
+    if (!directionsData.routes || directionsData.routes.length === 0) {
+      throw new Error('No route found between the specified locations');
+    }
+
+    const route = directionsData.routes[0];
+    if (!route.legs || route.legs.length === 0) {
+      throw new Error('Route has no legs');
+    }
+    const polyline = route.overview_polyline.points;
+
+    const points: { lat: number; lng: number }[] = [];
+    let index = 0;
+    let lat = 0;
+    let lng = 0;
+
+    while (index < polyline.length) {
+      let result = 0;
+      let shift = 0;
+      let byte: number;
+      do {
+        byte = polyline.charCodeAt(index++) - 63;
+        result |= (byte & 0x1f) << shift;
+        shift += 5;
+      } while (byte >= 0x20);
+      const deltaLat = result & 1 ? ~(result >> 1) : result >> 1;
+      lat += deltaLat;
+
+      result = 0;
+      shift = 0;
+      do {
+        byte = polyline.charCodeAt(index++) - 63;
+        result |= (byte & 0x1f) << shift;
+        shift += 5;
+      } while (byte >= 0x20);
+      const deltaLng = result & 1 ? ~(result >> 1) : result >> 1;
+      lng += deltaLng;
+
+      points.push({
+        lat: lat / 1e5,
+        lng: lng / 1e5,
+      });
+    }
+
+    const midpointIndex = Math.floor(points.length / 2);
+    const searchLocation = points[midpointIndex];
```

Comment:
There must exist an API for searching along a route instead of this. Research this online. Also, for all API operations for Google Maps, use the official Node API: https://github.com/googlemaps/google-maps-services-js

Reply:
Also check all other endpoints in these tools for whether official methods exist in the Node API.